WL-3502: Changes to how the Youtube widget works

nicholaswilson100 opened the pull request:
The Jira goes on a bit so a summary of the basic changes to how the widget currently works are:
1.   Just have one page of results showing, not 5.
2.   Make the Back, Next and page links look and work like Youtube's
3.   Disable the OK Button

The way the paging works below is that when a search is made, the results for page 1 are retrieved and with them also the pageTokens for the other pages (2-7).   When a page link is clicked, a click on the search button is then triggered using  the relevant pageToken.  

The currentPage is always stored so that when the Next button is clicked, we get the results for the pageToken after the one for currentPage.

nicholaswilson100 committed:
WL-3502: Changes to how the Youtube widget works

buckett commented on library/src/webapp/editor/ckextraplugins/youtube/js/service.js:
```diff
@@ -57,18 +83,36 @@ var YouTubeSearchService = function(options) {
-      data: prepareQueryParams({q: searchTerm}),
+      data: prepareQueryParams({q: searchTerm, pageToken: pageTokens[currentPage]}),
       dataType: 'json',
       async: false,
```
Shame the original code is not async.

nicholaswilson100 commented:
Yes async would have been better. I'd quite like to create (and do) a Jira for that...

nicholaswilson100 committed:
WL-3502 : Remove references to YouTubeSearchService in common code

nicholaswilson100 committed:
WL-3503 : Changes to the Vimeo ckeditor widget

nicholaswilson100 commented:
I'm pushing this branch  WL-3503 (Vimeo widget) onto this WL-3502 branch (Youtube) because there are quite a few changes that depend on it.

nicholaswilson100 commented:
The Vimeo changes are similar to the Youtube changes.  In summary, the main change is to make the page links look and work like Youtube as opposed to what it currently does which is just get 25 Vimeo videos and not give any opportunity to download more.  Also the Ok button is being removed here.

The way the paging works for Vimeo is a little different and simpler than Youtube.  For Vimeo, there is no need to to store pageTokens for each page beause Vimeo, unlike Youtube, offers the ability to query its API with a page number and it will then retrieve the right page of results for you.

I'm factoring out some common css which both Vimeo and Youtube use for its paging links (hence the new dialog.css file in 'common').

nicholaswilson100 commented:
There was also a reference to YoutubeSearchService in some common js which I've removed.

buckett commented on library/src/webapp/editor/ckextraplugins/common/js/itemsearch.js:
```diff
@@ -91,28 +106,41 @@ $.fn.itemSearch = function(options) {
+      }
+    }
+    else {
+        appendBackButton(nav, currentPage );
```
Shouldn't you be passing `iFrameId` in here?

nicholaswilson100 replied:
Yup good spot

buckett commented on library/src/webapp/editor/ckextraplugins/common/js/itemsearch.js:
```diff
@@ -91,28 +106,41 @@ $.fn.itemSearch = function(options) {
     }
 
     $(document).on('click', '.pageNum', function() {
       // find closest results container
```
I think you removed the closest selector.

nicholaswilson100 replied:
I've removed references to the localContainer in this method which was controlling the old js way of 'paging' the results.  I should have also removed this (now) redundant comment (or reworded it really)-  which I'll do.

buckett commented on library/src/webapp/editor/ckextraplugins/common/js/itemsearch.js:
```diff
@@ -91,28 +106,41 @@ $.fn.itemSearch = function(options) {
-
+      $.fn.itemSearch.currentPage = page;
+      var searchResultsIFrameId = $.grep(this.classList, function(v) {return v.indexOf('Iframe')!=-1;});
+      var searchButtonLink = document.getElementById(searchResultsIFrameId).contentWindow.document.forms[0].children[1];
```
I don't follow the code enough, but this looks live a very brittle way to get the search button as it's a very long change with multiple array indexes. What's the reason for this traversal?

nicholaswilson100 replied:
Yup you're right - it's very brittle. I'll find a better way.

nicholaswilson100 committed:
WL-3502 : Pass in iFrame, reword comment and simplify button selector

nicholaswilson100 committed:
WL-3500: Changes to CC image widget

nicholaswilson100 commented:
If it's okay, I'm also adding this Creative Commons Jira WL-3500 to this pull request as, like the Vimeo Jira above, it too relies on the new paging structure included in the initial Youtube Jira WL-3502 above.

For this Jira, there are a few changes and bug fixes. In summary:  the paging is changing to 25 per page (as well it working and looking like Youtube's paging structure), there was no title attribute on the image, some css had become unused (so I've removed it), and finally I've fixed a bug where if you change the image by using new search term the new image does not get saved.

I had to add some inline styling to the iframe in bind-creative-commons-image-search-to-dialog.js rather than in the css because the iframe's id is used elsewhere now with the new way paging works.

buckett commented:
Is this ready for review?

nicholaswilson100 commented:
Yup

buckett commented on library/src/webapp/editor/ckextraplugins/common/css/dialog.css:
```diff
@@ -0,0 +1,27 @@
+/* search results paging styles */
```
Is the use of pattern matching ID selectors so that you don't have to change the HTML?

nicholaswilson100 replied:
It's so I don't have to repeat myself in each widget's dialog.css.  The Youtube, Vimeo, and CreativeCommons widgets all now use the same paging styling and so share some css classes.

buckett replied:
It's just that normally if you're wanting to share styling across multiple nodes in different contexts you'd apply a class to them.

buckett commented on library/src/webapp/editor/ckextraplugins/common/js/itemsearch.js:
```diff
@@ -91,28 +106,41 @@ $.fn.itemSearch = function(options) {
+      $.fn.itemSearch.currentPage = page;
+      var searchButton = $('#' + this.id).contents().find('a');
+      searchButton.click();
+      e.preventDefault();
```
I don't see a `e` defined here, shouldn't the function have an argument?

buckett commented on library/src/webapp/editor/ckextraplugins/creative-commons-images/css/dialog.css:
```diff
@@ -91,7 +85,8 @@
 
-#creativeCommonsImageSearchResults .pagination {
+#creativeCommonsImage
+SearchResults .pagination {
```
This doesn't look correct, should there be a line break here?

buckett commented on library/src/webapp/editor/ckextraplugins/creative-commons-images/js/bind-creative-commons-image-search-to-dialog.js:
```diff
@@ -40,7 +40,7 @@ var BindCreativeCommonsImageSearchToDialog = function(path, dialog) {
   // the form without closing the dialog)
   var pushFormIntoIframe = function() {
-    var iframe = $('<iframe src="about:blank"></iframe>').attr('id', 'creativeCommonsImageSearchIframe');
+    var iframe = $('<iframe src="about:blank"></iframe>').attr('id', 'creativeCommonsImageSearchIframe').attr('style', 'height: 50px;width: 100%;');
```
You don't need to fix this, but for multiple options you can pass in a object  to the attr call and set them all in one go. Also for CSS changes you can use the css function.

nicholaswilson100 committed:
WL-3502: Various fixes to Youtube, Vimeo, Creative Commons

nicholaswilson100 commented:
In this 9349fe2 commit, I'm changing the css selectors to use classes rather than pattern matching as it's simpler, fixing a couple of bugs I hadn't previously noticed (one where a click handler is being attached more than once to the paging buttons so I'm removing any handlers first then attaching them, one where the 'Next' button wasn't working because I wasn't passing in the iFrameId),   removing a redundant line (e.preventDefault()), and passing in multiple attributes as an object rather than each separately.

nicholaswilson100 committed:
WL-3509: Changes and fixes to Solo widget

nicholaswilson100 commented:
In this 6436570 commit, I'm giving the Solo search results the same paging buttons/structure as Youtube, etc.  I'm moving the dialog higher up the screen and I'm removing the OK Button.  Some css is being moved inline as there is js logic that uses the css id.
